File: Python3/프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    order = sorted(priorities, reverse=True) #빠져나갈 순서
    order_index = 0
    answer = 0
    deq = deque()
    
    for x, y in enumerate(priorities):
        deq.append((x, y))
    
    index, val = None, None
    while deq:
        index, val = deq.popleft()
        if order[order_index] != val:
            deq.append((index, val))
        else:
            answer+= 1
            order_index += 1
            if index == location:
                break
    
    
    return answer

File: C++/프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    queue<vector<int>> q;
    for(int i = 0; i < priorities.size(); i++) {
        q.push({i, priorities[i]});
    }
    
    sort(priorities.begin(), priorities.end(), [&](int a, int b)->bool {
        return a > b;
    });
    int pIndex = 0; //우선순위 확인 인덱스
    
    vector<int> tmp;
    while(true) {
        tmp = q.front();
        q.pop();
        
        if(tmp[1] != priorities[pIndex]) {
            q.push(tmp);
        } else { //빠져나갈 수 있는 경우
            answer++; pIndex++;
            if(tmp[0] == location) {
                break;
            }
        }
    }
    
    return answer;
}